### PR TITLE
Preventing ClientSessionManager from calling onClinetShutdown multiple times 

### PR DESCRIPTION
**Pull request by Doxterpepper**
Saw this as a good easy bug to fix to get introduced to runelite. Is this a good solution?
#11354

### CODE REVIEW

**Comment by Adam-** [comment]
`sessionId` can also be null if a ping fails, so maybe that should also be fixed (as it would also throw without this bug). Maybe the correct way to fix this is to prevent the shutdown from happening twice?

**Comment by Doxterpepper** [comment]
So abort shutdown if the session ID is already null?

**Comment by Adam-** [comment]
The root of this problem is `shutdownClient` is being called >1 times I think. Not sure the best way to make it not do that. Completely unrelated to that, is L125 can set `sessionId = null` which means that it can be null in `onClientShutdown`, so I would just null check it there.

**Comment by deathbeam** [comment]
Simple isShuttingDown variable in ClientUI would suffice imo

**Comment by Doxterpepper** [comment]
> Simple isShuttingDown variable in ClientUI would suffice imo

That's what I was originally thinking too, but I felt like adding a new variable just for this added clutter. Especially when checking if sessionId is null does the same thing and prevents any other scenario where we'd call delete with a null session ID.

File: runelite-client/src/main/java/net/runelite/client/ClientSessionManager.java [hunk 1]
```diff
@@ -75,6 +75,11 @@ public void start()
 	@Subscribe
 	private void onClientShutdown(ClientShutdown e)
 	{
+		if (sessionId == null)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ClientSessionManager.java:78]: this null check should be down next to the `delete` call

File: runelite-client/src/main/java/net/runelite/client/ClientSessionManager.java [hunk 2]
```diff
@@ -81,7 +81,10 @@ private void onClientShutdown(ClientShutdown e)
 		{
 			try
 			{
-				sessionClient.delete(sessionId);
+				if (sessionId != null)
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ClientSessionManager.java:84]: There are potentially ~3 threads here (event, which comes in on EDT?), this thread, and the thread executing `ping`, so this is not thread safe. You should make a local reference to sessionId and then use that.

Comment (Doxterpepper) [reply runelite-client/src/main/java/net/runelite/client/ClientSessionManager.java]: Good catch, I hadn't thought of the threaded nature of this.

File: runelite-client/src/main/java/net/runelite/client/ClientSessionManager.java [hunk 3]
```diff
@@ -81,7 +81,11 @@ private void onClientShutdown(ClientShutdown e)
 		{
 			try
 			{
-				sessionClient.delete(sessionId);
+				UUID localUuid = sessionId;
+				if (localUuid != null)
+				{
+					sessionClient.delete(sessionId);
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/ClientSessionManager.java:87]: this should be `localUuid`

**Pull request merged**